inspirehep/records/api/literature.py

```diff
@@ -235,18 +235,21 @@ def get_modified_references(self):
             "deleted", False
         )
 
+        pids_latest = self.get_linked_pids_from_field("references.record")
+
         if changed_deleted_status:
-            return self.get_linked_pids_from_field("references.record")
+            return list(self.get_records_ids_by_pids(pids_latest))
 
-        ids_latest = set(self.get_linked_pids_from_field("references.record"))
         try:
-            ids_oldest = set(
+            pids_oldest = set(
                 self._previous_version.get_linked_pids_from_field("references.record")
             )
         except AttributeError:
-            return []
+            pids_oldest = []
+
+        pids_changed = set.symmetric_difference(set(pids_latest), pids_oldest)
 
-        return set.symmetric_difference(ids_latest, ids_oldest)
+        return list(self.get_records_ids_by_pids(list(pids_changed)))
```

Comment:
is the inner `list` really needed?

Reply:
yes, because otherwise I would be giving ```get_records_ids_by_pids``` a set. And ```get_records_ids_by_pids``` is doing the crazy things with the batches and is accessing the elements by index so it needs a list.